WarOOP/Models/Army.cs

```diff
@@ -0,0 +1,41 @@
+using System.Collections;
+using System.Threading.Channels;
+using WarOOP.Abstractions;
+
+namespace WarOOP.Models;
+
+public class Army : IArmy
+{
+    private List<Warrior> Units;
+
+    public Army()
+    {
+        Units = new List<Warrior>();
+    }
+    
+    public void AddUnits(Type type, int count)
+    {
+        for (int i = 0; i < count; i++)
+        {
+            Units.Add(Warrior.CreateWarrior(type));
+        }
+    }
+    
+    public bool HasUnits()
+    {
+        if (Units.Count > 0 && !Units[0].IsAlive)
+        {
+            Units.Remove(Units[0]);
```

Comment:
You just moved the logic to separate method. But it still has the same meaning. And this doesn't change anything. And again _HasUnits_ method's name doesn't sound like it should delete something. Try to think about logic without deleting any elements in collection